setup.py

```diff
@@ -31,6 +31,7 @@
         "tqdm",
         "huggingface_hub",
     ],
+    extras_require={"full": ["ctc-segmentation<1.8,>=1.6.6"],},  # noqa: E231
```

Comment:
I don't think we want that from scratch. Please check instead if it is installed, and prompt the users to do so if not (directly on the ctc_segmentation.py file). + Why do you need it, can't we code our own segmenter?

Reply:
I put this as a separate dependency, and didn't integrate the full code, because: On install, the CTC segmentation code needs to be cythonized (to be faster), and therefore requires a separate installation step (or, dynamic compilation with `pyximport`). I think that it is easier to install it from a Python package. See point 4 of the discussion above.

That said, I personally see also other advantages for using a package: Any change is also immediately available in other ASR frameworks (e.g., for testing). I highly welcome pull requests that improve or extend the code. Also, I prefer to maintain only one version of the code...

If you want to include the full code in speechbrain -
This includes `ctc_segmentation.py` and `ctc_segmentation_dyn.pyx` from:
https://github.com/lumaku/ctc-segmentation/tree/master/ctc_segmentation

So, because it is easier to install from Python package, I integrated it as an optional dependency.
For packages that are installed with setuptools, those optional requirements usually are defined in `extras_require`
as described in https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#optional-dependencies
Optional dependencies can then be installed together with the package by writing the dependency in brackets:
`pip install speechbrain[full]`.

Reply:
@TParcollet I was not entirely sure how you prefer to handle the optional dependency on `ctc-segmentation`.
Can you check if the current code is OK for you?

I additionally made a small change to how timing is calculated: I originally programmed this for espnet, that discards trailing audio input data; speechbrain frontend is a little bit different. I simply changed the calculation from integer to floating point. It's only a few ms, but it seems to be more accurate... If you retry with your previous short test audio file, maybe the alignments are better...?